src/workerd/api/crypto-impl-asymmetric.c++

```diff
@@ -103,6 +103,31 @@ public:
         "Asymmetric signing requires a private key.");
 
     auto type = lookupDigestAlgorithm(chooseHash(algorithm.hash)).second;
+    if (getAlgorithmName() == "RSASSA-PKCS1-v1_5") {
+      // RSASSA-PKCS1-v1_5 requires the RSA key to be at least as big as the digest size
+      // plus a 15 to 19 byte digest-specific prefix (see boringssl's RSA_add_pkcs1_prefix) plus 11
+      // bytes for padding (see RSA_PKCS1_PADDING_SIZE). For simplicity, require the key to be at
+      // least 32 bytes larger than the hash digest.
+      // Similar checks could also be adopted for more detailed error handling in verify(), but the
+      // current approach should be sufficient to avoid internal errors.
+      RSA& rsa = JSG_REQUIRE_NONNULL(EVP_PKEY_get0_RSA(getEvpPkey()), DOMDataError,
+          "Missing RSA key", tryDescribeOpensslErrors());
+      JSG_REQUIRE(EVP_MD_size(type) + 32 <= RSA_size(&rsa), DOMInvalidAccessError,
+          "key too small for signing with given digest");
+    } else if (getAlgorithmName() == "RSA-PSS") {
+      // Similarly, RSA-PSS requires keys to be at least the size of the digest and salt plus 2
+      // bytes, see https://developer.mozilla.org/en-US/docs/Web/API/RsaPssParams for details.
+      RSA& rsa = JSG_REQUIRE_NONNULL(EVP_PKEY_get0_RSA(getEvpPkey()), DOMDataError,
+          "Missing RSA key", tryDescribeOpensslErrors());
+      auto salt = JSG_REQUIRE_NONNULL(algorithm.saltLength, TypeError,
+          "Failed to provide salt for RSA-PSS key operation which requires a salt");
+      JSG_REQUIRE(salt >= 0, TypeError, "SaltLength for RSA-PSS must be non-negative (provided ",
+          salt, ").");
+      JSG_REQUIRE(EVP_MD_size(type) + 2 <= RSA_size(&rsa), DOMInvalidAccessError,
+          "key too small for signing with given digest");
+      JSG_REQUIRE(salt <= RSA_size(&rsa) - EVP_MD_size(type) - 2, DOMInvalidAccessError,
+          "key too small for signing with given digest and salt length");
```

Comment:
Let's make these all `DOMException`s for consistency.

Reply:
Not sure if you meant to mark this one resolved. There's still a `TypeError` there on line 152 that should be a `DOMException`

Reply:
Sorry, I misinterpreted your comment. Should be fixed now